shared/src/main/scala/squants/formatter/DefaultFormatter.scala

```diff
@@ -0,0 +1,22 @@
+package squants.formatter
+
+import squants.Quantity
+import squants.unitgroups.UnitGroup
+
+
+trait DefaultFormatter[A <: Quantity[A]] extends Formatter[A] {
+  def unitGroup: UnitGroup[A]
+
+  override def bestUnit(quantity: Quantity[A]): A = {
+    if (unitGroup.units.isEmpty)
```

Comment:
I'm not sure. `Set.empty` isn't a case object like `Nil`, so pattern matching on that won't work.